ckanext/string_to_location/tests/test_plugin.py

```diff
@@ -106,4 +152,31 @@ def _create_context(self):
             'user': user['name']
         }
 
-        return user, resource, context
\ No newline at end of file
+        return user, resource, context
+
+    def _create_correctly_formatted_csv(self, name):
+        data = [["Local authority", "Number"], ["Birmingham", 10]]
+        with open(name, 'wb') as f:
+            writer = csv.writer(f)
+            writer.writerows(data)
```

Comment:
Consider passing a `StringIO` object back here, rather than creating a file on the filesystem using that name. It behaves the same as a file object for our purposes:

```python
from StringIO import StringIO
output_buffer = StringIO()
writer = csv.writer(output_buffer)
writer.writerows(data)

# Elsewhere
upload = cgi.FieldStorage()
upload.filename = 'test.csv'
upload.file = outputbuffer
```